### PR TITLE
[MRG] Fix `sourmash prefetch` to work when db scaled is larger than query scaled

### PR DESCRIPTION
**Pull request by ctb**
This PR fixes a problem where `sourmash prefetch` fails to properly downsample the query sketch as needed.

TODO:

- [x] do we need to check against other database types?
- [x] fix assert in re threshold calculation; allow skip? maybe test this more.
- [x] notify user of scaled value for final summary sketches
- [x] make sure that output sketches are at appropriate scaled

### CODE REVIEW

File: src/sourmash/commands.py [hunk 1]
```diff
@@ -1223,9 +1222,20 @@ def prefetch(args):
...
+            if ident_mh.scaled != common_scaled:
+                ident_mh = ident_mh.downsample(scaled=common_scaled)
+            if noident_mh.scaled != common_scaled:
+                noident_mh = noident_mh.downsample(scaled=common_scaled)
+
             # track found & "untouched" hashes.
-            match_mh = match.minhash.downsample(scaled=query.minhash.scaled)
-            ident_mh += query.minhash & match_mh.flatten()
+            ident_mh += query_mh & match_mh.flatten()
             noident_mh.remove_many(match.minhash)
```
Comment (bluegenes) [inline src/sourmash/commands.py]: is there a benefit to using `match.minhash` vs `match_mh` here?

File: tests/test_prefetch.py [hunk 2]
```diff
@@ -52,6 +52,67 @@ def test_prefetch_select_query_ksize(runtmp, linear_gather):
     assert 'of 4476 distinct query hashes, 4476 were found in matches above threshold.' in c.last_result.err
 
 
+def test_prefetch_subject_scaled_is_larger(runtmp, linear_gather):
+    # test prefetch where query and subject db both have multiple ksizes
```
Comment (bluegenes) [inline tests/test_prefetch.py]:
```suggestion
    # test prefetch where subject scaled is larger
```

File: tests/test_prefetch.py [hunk 3]
```diff
@@ -52,6 +52,67 @@ def test_prefetch_select_query_ksize(runtmp, linear_gather):
...
+    print(c.last_result.err)
+
+    assert c.last_result.status == 0
+    assert 'total of 8 matching signatures.' in c.last_result.err
+    assert 'of 48 distinct query hashes, 48 were found in matches above threshold.' in c.last_result.err
+    assert 'final scaled value (max across query and all matches) is 10000' in c.last_result.err
+
+
+def test_prefetch_subject_scaled_is_larger_outsigs(runtmp, linear_gather):
+    # test prefetch output sigs
```
Comment (bluegenes) [inline tests/test_prefetch.py]:
```suggestion
    # test prefetch where subject scaled is larger -- output sigs
```

**Review by bluegenes — APPROVED** [review]
a couple minor things, otherwise lgtm!

File: src/sourmash/commands.py (hunk 1 above)
Comment (ctb) [reply src/sourmash/commands.py]: nope. fixed in f3b1a7af. good catch.

**Pull request merged**